docs/vhost-user-net-testing.md

```diff
@@ -0,0 +1,130 @@
+             |               |          |                                       |          |               |           
+          +--+---------------------------------------------------------------------------------------------+--+        
+          |  |                                       hugepage                                              |  |        
+          |  +---------------------------------------------------------------------------------------------+  |        
+          |                                                                                                   |        
+          |                                              host                                                 |        
+          |                                                                                                   |        
+          +---------------------------------------------------------------------------------------------------+        
+```
+### Prerequisites
+
+Prior to run the test, the following steps need to be done, [vhost-user-ovs-dpdk](https://wiki.qemu.org/Documentation/vhost-user-ovs-dpdk) is a good reference which tells how to do.
+- Enable hugepage
+- Install kernel-devel
+- Build DPDK from source
+- Build OVS from source
+
+### Test
+The test will run with mq support enabled, 2 queue pairs defined for both OVS and virtual machine. Here is the details how we can run the test.
+
+_Setup OVS_
+
+ovs_test.sh is created to setup and start OVS. OVS will provide the dpdkvhostuser type port which is as server mode.
+```
+#!/bin/bash
+
+mkdir -p /var/run/openvswitch
+modprobe openvswitch
+killall ovsdb-server ovs-vswitchd
+rm -f /var/run/openvswitch/vhost-user*
+rm -f /etc/openvswitch/conf.db
+export DB_SOCK=/var/run/openvswitch/db.sock
+ovsdb-tool create /etc/openvswitch/conf.db /usr/share/openvswitch/vswitch.ovsschema
+ovsdb-server --remote=punix:$DB_SOCK --remote=db:Open_vSwitch,Open_vSwitch,manager_options --pidfile --detach
+ovs-vsctl --no-wait init
+ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-lcore-mask=0xf
+ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-socket-mem=1024
+ovs-vsctl --no-wait set Open_vSwitch . other_config:dpdk-init=true
+ovs-vsctl --no-wait set Open_vSwitch . other_config:pmd-cpu-mask=0xf
+ovs-vswitchd unix:$DB_SOCK --pidfile --detach --log-file=/var/log/openvswitch/ovs-vswitchd.log
+ovs-vsctl add-br ovsbr0 -- set bridge ovsbr0 datapath_type=netdev
+ovs-vsctl add-port ovsbr0 vhost-user1 -- set Interface vhost-user1 type=dpdkvhostuser
+ovs-vsctl add-port ovsbr0 vhost-user2 -- set Interface vhost-user2 type=dpdkvhostuser
+ovs-vsctl set Interface vhost-user1 options:n_rxq=2
+ovs-vsctl set Interface vhost-user2 options:n_rxq=2
+```
+Run ovs_test.sh
+```bash
+./ovs_test.sh
+```
+
+_Launch the VMs_
+
+VMs will be as client to connect to the dpdkvhostuser port socket.
+```bash
+# From one terminal. Run as root, it's allowed to use tap interface without additional settings.
```

Comment:
I'd prefer if you were mentioning that setting the right capabilities to the cloud-hypervisor binary will be enough. It's better if we avoid saying the VMM must be run as root.